### PR TITLE
Refactored weapons.bin into weapons.json for extensibility and readability.

### PR DESCRIPTION
**Pull request by gir489returns**
Added weapon attachments scraping from the meta files (currently is missing a lot of attachments, more than half, requires RPF reading refactoring to fix.)
Added Ammunation to Self -> Weapons, because it's vital you protect yourself, the patriotic way.

Please conduct code review and address issue of RPF loader and weapon_file::version_info not populating properly.

### CODE REVIEW

File: src/services/gta_data/gta_data_service.cpp [hunk 1]
```diff
@@ -352,6 +414,8 @@ namespace big
 
 							const char* category = "";
 
+							int attachment_counter = 0;
```
Comment (gir489returns) [inline src/services/gta_data/gta_data_service.cpp:417]: Useless counter, remove.

File: src/views/self/view_weapons.cpp [hunk 2]
```diff
@@ -176,5 +176,104 @@ namespace big
...
+					selected_weapon_hash = weapon.second.m_hash;
+				}
+				if (is_selected)
+					ImGui::SetItemDefaultFocus();
+			}
+			ImGui::EndCombo();
+		}
+		ImGui::PopItemWidth();
+		ImGui::SameLine();
+		if (ImGui::Button("Give Weapon"))
```
Comment (xiaoxiao921) [inline src/views/self/view_weapons.cpp]: All of these ImGui::Button with queue_job inside the if can be replaced with components::button(text, callback) calls

Comment (gir489returns) [reply src/views/self/view_weapons.cpp:201]: Thanks. I just straight copied them from my fork from BBv2, didn't know you had wrappers.

File: src/views/self/view_weapons.cpp [hunk 3]
```diff
@@ -198,18 +198,14 @@ namespace big
 		}
 		ImGui::PopItemWidth();
 		ImGui::SameLine();
-		if (ImGui::Button("Give Weapon"))
+		if (components::button("Give Weapon"))
```
Comment (xiaoxiao921) [inline src/views/self/view_weapons.cpp]: https://github.com/YimMenu/YimMenu/blob/cb9205e478c16e6b4686f7fbdee04d0046388c2f/src/views/self/view_weapons.cpp#L93

this is what I meant, you need to use the overload with the callback in it for the natives to work

**Comment by gir489returns** [comment]
This PR should be complete now, ready for full review.

**Comment by gir489returns** [comment]
The new RPF loader introduced with #1635 gives me less components and less weapons than I previously had.

COMPONENT_FLASHLIGHT_LIGHT, COMPONENT_EMPLAUNCHER_CLIP_01, COMPONENT_HEAVYRIFLE_CAMO1, COMPONENT_HEAVYRIFLE_CLIP_01, COMPONENT_HEAVYRIFLE_CLIP_02, COMPONENT_HEAVYRIFLE_SIGHT_01, COMPONENT_MACHINEPISTOL_CLIP_01, COMPONENT_MACHINEPISTOL_CLIP_02, VEHICLE_WEAPON_BOMB, VEHICLE_WEAPON_BOMB_CLUSTER, VEHICLE_WEAPON_BOMB_GAS, VEHICLE_WEAPON_BOMB_INCENDIARY, WEAPON_FERTILIZERCAN, WEAPON_HEAVYRIFLE are all missing.

COMPONENT_FLASHLIGHT_LIGHT is part of mphalloween.
COMPONENT_EMPLAUNCHER_CLIP_01, COMPONENT_HEAVYRIFLE_CAMO1, COMPONENT_HEAVYRIFLE_CLIP_01, COMPONENT_HEAVYRIFLE_CLIP_02, COMPONENT_HEAVYRIFLE_SIGHT_01 are all part of mpsecurity
COMPONENT_MACHINEPISTOL_CLIP_01, COMPONENT_MACHINEPISTOL_CLIP_02 are part of mplowrider

**Comment by gir489returns** [comment]
#1641 addresses the regression issue introduced with #1635.

File: src/services/gta_data/gta_data_service.cpp [hunk 4]
```diff
@@ -145,7 +160,26 @@ namespace big
 	{
 		m_peds_cache.load();
 		m_vehicles_cache.load();
-		m_weapons_cache.load();
+
+		auto weapons_file = big::g_file_manager->get_project_file("./cache/weapons.json");
```
Comment (Yimura) [inline src/services/gta_data/gta_data_service.cpp]: Unnecessary `big::` namespace addition.

**Review by Yimura — CHANGES_REQUESTED** [review]

File: src/services/gta_data/gta_data_service.cpp (hunk 4 above)
Comment (gir489returns) [reply src/services/gta_data/gta_data_service.cpp]: Removed.

**Review by Yimura — APPROVED** [review]
Seems I missed another `big::g_file_manager`, I'll fix that up in my rewrite of the file mgr PR.

**Comment by gir489returns** [comment]
> Seems I missed another `big::g_file_manager`, I'll fix that up in my rewrite of the file mgr PR.

Heh. Guess where I copied it from. 😂

**Pull request merged**